SqlAssignments.sql

```diff
@@ -0,0 +1,1846 @@
+| Rome    |        200 |
+| SanJose |        300 |
++---------+------------+
+4 rows in set (0.00 sec)
+
+
+
+
+**************************************************************************************************************************************
+
+
+
+
+Que 41 ->Write a query that calculates the amount of the salesperson’s commission on each order by a customer with a rating above 100.00.
+
+Query  ->SELECT S.snum,S.sname,S.comm*O.amt AS total_amt 
+       ->FROM salespeople S 
+       ->INNER JOIN Customers C ON C.snum=S.snum 
+       ->INNER JOIN Orders O ON C.cnum=O.cnum  
+       ->WHERE C.rating > 100;
++------+---------+--------------------+
+| snum | sname   | total_amt          |
++------+---------+--------------------+
+| 1003 | AxelRod |            17132.3 |
+| 1002 | Serres  |  67085.84999999999 |
+| 1002 | Serres  |             984.75 |
+| 1002 | Serres  | 17029.350000000002 |
+| 1007 | Rifkin  |             280.35 |
+| 1007 | Rifkin  |            16472.4 |
++------+---------+--------------------+
+6 rows in set (0.00 sec)
+
+
+
+**************************************************************************************************************************************
+
+
+
+Que 42 -> Count the customers with ratings above SanJose’s average.
+
+Query  ->SELECT COUNT(city) AS total_count 
+       ->FROM Customers 
+       ->WHERE rating > (SELECT AVG(rating)                 
+                         FROM Customers                 
+                         WHERE city='SanJose');
++-------------+
+| total_count |
++-------------+
+|           2 |
++-------------+
+1 row in set (0.00 sec)
+
+
+
+**************************************************************************************************************************************
+
+Que 43 ->Find all salespeople that are located in either Barcelona or London
+
+Query  ->SELECT snum,sname,city
+       ->FROM salespeople
+       ->WHERE city='Barcelona' OR city='London';
++------+--------+-----------+
+| snum | sname  | city      |
++------+--------+-----------+
+| 1001 | Peel   | London    |
+| 1004 | Motika | London    |
+| 1007 | Rifkin | Barcelona |
+| 1008 | Fran   | London    |
++------+--------+-----------+
+4 rows in set (0.00 sec)
+
+
+**************************************************************************************************************************************
+
+
+Que 44 ->Find all salespeople with only one customer
+
+Query  ->SELECT snum
+       -> FROM Customers
+       -> GROUP BY snum
+       -> HAVING COUNT(snum)<=1;
++------+
+| snum |
++------+
+| 1003 |
+| 1004 |
+| 1007 |
++------+
+3 rows in set (0.00 sec)
+
+
+
+**************************************************************************************************************************************
+
+
+
+
+Que 45 ->Write a query that joins the Customer table to itself to find all pairs or customers served by a single salesperson.
+
+Query  ->SELECT C1.cname AS Cname1,C2.cname AS Cname2,S.sname  
```

Comment:
I think this query is not written as the correct one. Please rethink it.

Reply:
Sure sir